src/Microsoft.PowerShell.Commands.Management/commands/management/TestConnectionCommand.cs

```diff
@@ -389,8 +389,8 @@ private void ProcessTraceroute(string targetNameOrAddress)
                 {
                     try
                     {
-#if !UNIX
-                        if (ResolveDestination.IsPresent && routerName == string.Empty)
+                        if (routerName == string.Empty
+                            || ResolveDestination.IsPresent && routerName == hopAddressString)
```

Comment:
Why do we need InitProcessPing() in the `for` cycle if we ping the same address?

Reply:
Mainly for -ResolveDestination purposes, to retrieve the DNS name for the hop point. Note that it is skipped if there is already a name retrieved for that hop point.

Reply:
We call InitProcessPing() for the _same hop_ in the cycle. I suggest to move out the call out of the cycle.

Reply:
Moved the InitProcessPing() call to just before the loop. 👍